water_mark/water_mark.py

```diff
@@ -0,0 +1,284 @@
+
+
+def copy_vv_vh_to_inputs(outDir, dataDict) -> None:
+    inputsPath = os.path.join(os.getcwd(), 'inputs')
+    for sar, vvvhband in dataDict.items():
+        shutil.copy(os.path.join(inputsPath, vvvhband[0]),
+                    os.path.join(os.getcwd(), outDir, sar))
+        shutil.copy(os.path.join(inputsPath, vvvhband[1]),
+                    os.path.join(os.getcwd(), outDir, sar))
+
+
+def make_masks(outDir, dataDict) -> None:
+    count = 1
+    for sar, vvvhband in dataDict.items():
+        inputPath = os.path.join(os.getcwd(), 'inputs')
+        renamePath = os.path.join(os.getcwd(), outDir, sar)
+        print(f"{count}")
+        idw_main(os.path.join(inputPath, vvvhband[0]),
+                 os.path.join(inputPath, vvvhband[1]))
+        count += 1
+        # rename 'mask-0.tif' -> 'Mask_<sar>.tif'
+        os.rename('mask-0.tif',
+                  os.path.join(renamePath, 'Mask_'+sar+'.tif'))
+
+
+def tile(outDir, tifName, sar, mxmTileSize, isMask) -> None:
+    label = 'temp'
+    if isMask:
+        label = ''
+    else:
+        label = 'Image'
+    tif = os.path.join(outDir, sar, tifName)
+    tifData = gdal.Open(tif)
+    xStep, yStep = mxmTileSize, mxmTileSize
+    xSize, ySize = tifData.RasterXSize, tifData.RasterYSize
+    count = 0
+    for x in range(0, xSize, xStep):
+        for y in range(0, ySize, yStep):
+            # fileName = '<Image|Mask>_<sar>_<0-9+>.tif'
+            fileName = f"{label}_{tifName[:-4]}_{count}.tif"
+            gdal.Translate(os.path.join(outDir, sar, fileName),
+                           tif,
+                           srcWin=[x, y, xStep, yStep],
+                           format="GTiff")
+            count += 1
+
+
+def tile_vv_vh_mask(outDir, mxmTileSize) -> None:
+    for sar in os.listdir(os.path.join(outDir)):
+        for tifName in os.listdir(os.path.join(outDir, sar)):
+            if tifName.endswith('VV.tif'):
+                tile(outDir, tifName, sar, mxmTileSize, False)
+            elif tifName.endswith('VH.tif'):
+                tile(outDir, tifName, sar, mxmTileSize, False)
+            elif tifName.startswith('Mask'):
+                tile(outDir, tifName, sar, mxmTileSize, True)
+
+
+def main():
+    # set tile size manually here (comment out parser block below)
+    # mxmTileSize = 512
+
+    # cmd line argumnet parser for tile size
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-s', type=int, default=512)
+    parser.add_argument('-worldMask, type=bool, default=False')
+    args = parser.parse_args()
+    mxmTileSize = args.s
+    worldMask = args.worldMask
+
+    outDir = 'syntheticTriainingData'+date.isoformat(date.today())  # outDir
+    data = make_database()
+
+    make_output_dir(outDir, data)
+    if worldMask:
+        cut_worldMask_to_sar()
+        move_sar_mask_to_outDir()
+        cut_sar_to_tiles()
+        clip_mask_to_sar()
+        trim_masks()
+        reclassify_mask()
+        clean_up()
+        sys.exit()
```

Comment:
```suggestion
        return
```

If this gets called from some other program you don't want it to kill the other program